mne/encoding/model.py

```diff
@@ -0,0 +1,154 @@
+            An object for preprocessing / transforming the output data before
+            the call to `est`. If None, no preprocessing will occur.
+
+        References
+        ----------
+        .. [1] Smith, N. J., & Kutas, M. (2015). Regression-based estimation of
+               ERP waveforms: II. Non-linear effects, overlap correction, and
+               practical considerations. Psychophysiology, 52(2), 169-189.
+        .. [2] Theunissen, F. E. et al. Estimating spatio-temporal receptive
+               fields of auditory and visual neurons from their responses to
+               natural stimuli. Network 12, 289-316 (2001).
+        .. [3] Willmore, B. & Smyth, D. Methods for first-order kernel
+               estimation: simple-cell receptive fields from responses to
+               natural scenes. Network 14, 553-77 (2003).
+        """
+        self.delays = np.asarray(delays)
+        self.n_delays = len(self.delays)
+        self.est = _check_estimator(est)
+        self.preproc_y = _check_preproc(preproc_y)
+        self.preproc_x = _check_preproc(preproc_x)
+
+    def fit(self, X, y=None):
+        """Fit the model.
+
+        Fits a receptive field model. Model results are stored as attributes.
+
+        Parameters
+        ----------
+        raw : instance of MNE Raw
+            The data on which we want to fit a regression model.
+        events : array, shape (n_events, 3)
+            An MNE events array specifying indices for event onsets
+        event_id : dictionary | None
+            A dictionary of (event_name: event_int) pairs, corresponding to the
+            mapping from event name strings to the 3rd column of events.
+        continuous : array, shape (n_feats, n_times)
+            Continuous input features in the regression.
+        continuous_names : array, shape (n_feats,) | None
+            Names for the input continuous variables.
+        picks : array, shape (n_picks,) | None
+            Indices for channels to use in model fitting. If None, all channels
+            will be fit.
+
+        Attributes
+        ----------
+        coefs_ : array, shape (n_channels, n_features)
+            The coefficients fit on each row of X
+        coef_names : array, shape (n_features,)
+            A list of coefficient names, useful for keeping track of time lags
+        """
+        if y is None:
+            raise ValueError('Must supply an output variable `y`')
+        if X.shape[-1] != y.shape[-1]:
+            raise ValueError('X and y must have last dimension same length.')
+        n_chs, n_times = y.shape
+
+        # Prepare the input feature matrix
+        if self.preproc_x is not None:
+            X = self.preproc_x.fit_transform(X)
+        if self.preproc_y is not None:
+            y = self.preproc_y.fit_transform(y)
+        n_features = X.shape[0]
+
+        # Fit the model and assign coefficients
+        self.coef_ = np.zeros([n_chs, n_features])
+        for ii in range(n_chs):
```

Comment:
Hmmm, in my case it's not too long. Off the top of my head, using Ridge regression with ~800 regressors, 20 CV iterations, downsampled to 100 Hz so maybe 20k-30k timepoints, it'll take maybe a minute or two per channel.

So I could fit all the channels in like an hour or two (or less if I didn't do so much cross validation), but IME once you're at the point of fitting multiple regression models w/ cross validation etc, one per channel, then it's better to use some kind of distributed computing system. E.g., I use the SunGridEngine to distribute my jobs so fitting all of those models takes 3-4 minutes.

Though perhaps it's worth systematically varying parameters and seeing how it affects the fit. I've tested Ridge with a bunch of fitting algorithms and there are big differences. I've also done the regression with stochastic gradient descent and it's a *lot* faster, but there are more knobs to tweak as well.